rimage/depth_map.go

```diff
@@ -99,23 +99,17 @@ func (dm *DepthMap) At(x, y int) color.Color {
 func (dm *DepthMap) ColorModel() color.Model { return color.Gray16Model }
 
 // SubImage returns a cropped image of the original DepthMap from the given rectangle.
-func (dm *DepthMap) SubImage(r image.Rectangle) *DepthMap {
+func (dm *DepthMap) SubImage(rect image.Rectangle) *DepthMap {
+	r := rect.Intersect(dm.Bounds()).Sub(dm.Bounds().Min)
 	if r.Empty() {
-		return &DepthMap{}
+		return NewEmptyDepthMap(0, 0)
```

Comment:
Nit: If you really want to make sure no one is using `&DepthMap{}` you could un-export `DepthMap` and generalize `NewEmptyDepthMap` to something like `NewDepthMap`.  But I'm not sure if using `NewEmptyDepthMap` is an optimization - it pre-allocates the space it needs - or defensive - it ensures future random accesses won't segfault. But that's out of the scope for this so just a nit/thought.

Reply:
This is more because I don't want the data field of DepthMap to have a nil value, I want it to be initialized with an empty slice